test/parallel/test-http2-compat-expect-handling.js

```diff
@@ -0,0 +1,46 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const expectValue = 'meoww';
+
+const server = http2.createServer(common.mustNotCall());
+
+server.once('checkExpectation', common.mustCall((req, res) => {
+  assert.strictEqual(req.headers['expect'], expectValue);
+  res.statusCode = 417;
+  res.end();
+}));
+
+server.listen(0, common.mustCall(() => nextTest(2)));
+
+function nextTest(testsToRun) {
+  if (!testsToRun) {
+    return server.close();
+  }
```

Comment:
This can use the new `Countdown` utility under `common`... e.g.

```js
const Countdown = require('../common/countdown');
const countdown = new Countdown(2, () => console.log('done'));
countdown.dec();
countdown.dec();
```